src/templates/fundraising/index.js

```diff
@@ -1,9 +1,131 @@
+/* eslint-disable react/prop-types */
+import React from 'react'
+import { Decimal } from 'decimal.js'
+import {
+  ClaimDomainScreen,
+  FundraisingScreen,
+  KnownAppBadge,
+  ReviewScreen,
+  TokensScreen,
+  VotingScreen,
+} from '../kit'
+import BoardInfo from './components/BoardInfo'
+import ShareInfo from './components/ShareInfo'
 import header from './header.svg'
 import icon from './icon.svg'
 
+function BN(value) {
+  return new Decimal(value)
+}
+
+const PPM = BN(10).pow(BN(6))
+const oneDay = BN(24).mul(BN(3600))
+const oneMonth = BN(30).mul(oneDay)
+const oneDAI = BN(10).pow(BN(18))
+const onePercent = BN(10).pow(BN(16))
+const onePercentPPM = BN(10).pow(BN(4))
+const oneBlock = BN(15)
+const one = BN(1)
+
+function completeDomain(domain) {
+  return domain ? `${domain}.aragonid.eth` : ''
+}
+
+function extractVotingSettings(voting) {
+  const support = onePercent.mul(BN(voting.support)).toFixed(0)
+  const quorum = onePercent.mul(BN(voting.quorum)).toFixed(0)
+  const duration = BN(voting.duration).toFixed(0)
+
+  return [support, quorum, duration]
+}
+
+function extractFundraisingAppsSettings(fundraising) {
+  const goal = oneDAI.mul(BN(fundraising.targetGoal)).toFixed(0)
+  const period = oneDay.mul(BN(fundraising.fundingPeriod)).toFixed(0)
+  const exchangeRate = PPM.div(BN(fundraising.presalePrice)).toFixed(0)
+  const vestingCliffPeriod = oneDay.mul(BN(fundraising.cliffPeriod)).toFixed(0)
+  const vestingCompletePeriod = oneDay
+    .mul(BN(fundraising.vestingSchedule))
+    .toFixed(0)
+  const maximumMonthlyUpdates = onePercent
+    .mul(BN(fundraising.maximumMonthlyUpdates))
+    .toFixed(0)
+  const supplyOfferedPct = onePercentPPM
+    .mul(BN(fundraising.tokensOffered))
+    .toFixed(0)
+  const fundingForBeneficiaryPct = onePercentPPM
+    .mul(BN(fundraising.projectFunding))
+    .toFixed(0)
+
+  return {
+    goal,
+    period,
+    exchangeRate,
+    vestingCliffPeriod,
+    vestingCompletePeriod,
+    maximumMonthlyUpdates,
+    supplyOfferedPct,
+    fundingForBeneficiaryPct,
+  }
+}
+
+function extractCollateralizationSettings(fundraising) {
```

Comment:
Would be great to add a few comments to this particular function's calculations. The names will likely not be as descriptive three months from now ;)

Reply:
Ping @osarrouy, it would probably be 1000.01% more correct if you are the one doing this :smile: